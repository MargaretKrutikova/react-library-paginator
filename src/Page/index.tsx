import * as React from "react";
import * as PropTypes from "prop-types";

export type PropType = {
  isActive?: boolean;
  isDisabled: boolean;
  page: number;
  children?: React.ReactNode;
  onPageChange: (page: number) => void;
};

class Page extends React.PureComponent<PropType> {
  static propTypes = {
    isActive: PropTypes.bool,
    isDisabled: PropTypes.bool,
    page: PropTypes.number.isRequired,
    onPageChange: PropTypes.func.isRequired
  };
  setPage = (event: React.MouseEvent<HTMLElement>, page: number) => {
    if (event) {
      event.preventDefault();
    }
    if (!this.props.isDisabled && this.props.isActive) {
      this.props.onPageChange(page);
    }
  };
  getPageClass = () => {
    const classes = ["page-item"];
    this.props.isActive && classes.push("active");
    this.props.isDisabled && classes.push("disabled");

    return classes.join(" ");
  };
  render() {
    const { page } = this.props;

    return (
      <li className={this.getPageClass()}>
        <a
          href="#"
          className="page-link"
          onClick={event => this.setPage(event, page)}
        >
          {this.props.children || page}
        </a>
      </li>
    );
  }
}

export default Page;
