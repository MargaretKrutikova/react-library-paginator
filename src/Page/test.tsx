import * as React from "react";
import * as Enzyme from "enzyme";
import Page, { PropType } from ".";

describe("Page", () => {
  it("it changes classes if isActive or isDisabled props change", () => {
    const testProps: PropType = {
      isActive: false,
      isDisabled: false,
      page: 4,
      // tslint:disable-next-line:no-empty
      onPageChange: (_: number) => {}
    };
    const component: Enzyme.ShallowWrapper = Enzyme.shallow(
      <Page {...testProps} />
    );

    expect(component).toMatchSnapshot();

    // change props and rerender
    component.setProps({ isActive: true });
    expect(component).toMatchSnapshot();

    // change props and rerender
    component.setProps({ isActive: false });
    component.setProps({ isDisabled: true });
    expect(component).toMatchSnapshot();
  });
});
