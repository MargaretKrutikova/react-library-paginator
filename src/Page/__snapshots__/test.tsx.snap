// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page it changes classes if isActive or isDisabled props change 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={false}
    isDisabled={false}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="page-link"
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "page-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "className": "page-link",
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "page-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "className": "page-link",
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Page it changes classes if isActive or isDisabled props change 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={true}
    isDisabled={false}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="page-link"
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "page-item active",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "className": "page-link",
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "page-item active",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "className": "page-link",
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Page it changes classes if isActive or isDisabled props change 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={false}
    isDisabled={true}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="page-link"
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "page-item disabled",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "className": "page-link",
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "page-item disabled",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "className": "page-link",
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
